controllers/shipwrightbuild_controller.go

```diff
@@ -69,7 +79,75 @@ func (r *ShipwrightBuildReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	logger := r.Logger.WithValues("namespace", req.Namespace, "name", req.Name)
 	logger.Info("Starting resource reconciliation...")
 
-	// retrieving the ShipwrightBuild instance requested for reconciliation
+	// See if tekton is there
+	_, tektonAPIErr := r.CRDClient.CustomResourceDefinitions().Get(ctx, "taskruns.tekton.dev", metav1.GetOptions{})
+	tektonOpCRD, tektonOPErr := r.CRDClient.CustomResourceDefinitions().Get(ctx, "tektonconfigs.operator.tekton.dev", metav1.GetOptions{})
+	switch {
+	case tektonOPErr == nil && tektonAPIErr == nil:
+		logger.Info("Tekton is installed, and the Tekton Operator is also installed.")
+	case tektonAPIErr == nil && tektonOPErr != nil:
+		logger.Info("Tekton has been installed without use of its associated operator.")
+	case errors.IsNotFound(tektonAPIErr) && errors.IsNotFound(tektonOPErr):
+		//TODO should we error out here or proceed and give the user the opportunity of install Tekton after Shipwright?
+		logger.Info("Evidence of neither required Tekton APIs nor the Tekton Operator being installed can be found.  Will proceed with the Shipwright installation attempt, but Shiipwright will not be functional without Tekton")
+	case tektonAPIErr != nil && tektonOPErr == nil:
+		if tektonOpCRD.Labels == nil {
+			retErr := fmt.Errorf("the CRD TektonConfig does not have labels set, inclding its version")
+			logger.Error(retErr, "Problem confirming Tekton Operator version")
+			return RequeueWithError(retErr)
+		}
+		value, exists := tektonOpCRD.Labels["version"]
+		if !exists {
+			retErr := fmt.Errorf("the CRD TektonConfig does not have labels set, inclding its version")
+			logger.Error(retErr, "Problem confirming Tekton Operator version")
+			return RequeueWithError(retErr)
+		}
+		// replace v0 with v1 to make k8s version check happy
+		if strings.HasPrefix(value, "v0") {
+			value = strings.ReplaceAll(value, "v0", "v1")
+		}
+		version, err := version.ParseSemantic(value)
```

Comment:
Curious why this is the case - `v0.6.1` is a perfectly valid semver.

Reply:
it has been that way since this utility landed in apimachinery:  see https://github.com/kubernetes/apimachinery/commit/0e7924a69fe3c0d366ed9cd674fac70c074386b2#diff-7f2dfde88efef53aa3c8aba302b41cb5b2b9037095700fa5a3ecb461a9123374 and the godoc for `ParseGeneric`.  There at least is explicit intent there.

I *think* the code was in k8s/k8s before that, though I did not find it immediately, and claimed due diligence 10 minutes into that.

If your curiosity prompts you to continue the search and find out @adambkaplan do let us know :-)

Reply:
First digit 0 is acceptable. _Leading_ zero (such as `01`) is not.

See https://github.com/kubernetes/apimachinery/blob/master/pkg/util/version/version_test.go#L76 and https://github.com/kubernetes/apimachinery/blob/master/pkg/util/version/version_test.go#L148